src/main/java/de/retest/recheck/configuration/ProjectConfiguration.java

```diff
@@ -40,10 +40,12 @@ public static ProjectConfiguration getInstance() {
 		return instance;
 	}
 
+	public Optional<Path> getProjectConfigFolder() {
```

Comment:
Find it hard to distinguish between `getProjectConfigFolder()` and `findProjectConfigFolder()`, only the return types differ.

As the latter potentially throws an exception (like `Optional#get()`) and the former doesn't, how about:

* `getProjectConfigFolder()` => `searchProjectConfigFolder()` returns `Optional`.
* `findProjectConfigFolder()` => `getProjectConfigFolder()` potentially throws an exception.

I'm not sure, ideas?

Reply:
I would have removed it, or at least made it private. But I could not be bothered to look at where it is used...

Reply:
The `get...` is related to `ProjectRootFinderUtil.getProjectRoot()` where also an `Optional` is returned.

Reply:
OK, I see, makes sense!